### DISCUSSION TITLE
[latex] Highlighting comments insite math regions

### DISCUSSION THREAD

**Discussion by gi1242** [nvim-treesitter/nvim-treesitter#6038] [Q&A]
Hi All,

In Latex files, is it possible for comments inside the file to be highlighted as comments. I'm attaching a screenshot of what it looks like now:

![image](https://github.com/nvim-treesitter/nvim-treesitter/assets/1884258/0966deff-ffcb-46ac-9155-7ea1bf10e0c8)

For the comment that is highlighted correctly, `:Inspect` shows:

    Treesitter
      - @markup.math.latex links to Special latex
      - @spell.latex links to @spell latex
      - @comment.latex links to Comment latex
      - @spell.latex links to @spell 

For the comment that is highlighted incorrectly, `:Inspect` shows:

    Treesitter
      - @comment.latex links to Comment latex
      - @spell.latex links to @spell latex
      - @markup.math.latex links to Special latex

I poked around the source files but couldn't find a good way to fix it.

Is this a bug? Or is it by design...?

Also is this the right place to ask?

Thanks in advance,

GI

**Comment by clason** [comment]
That's an ordering issue; probably a regression due to a recent PR @ribru17

**Reply by gi1242** [reply]
Good to hear it's not by design 😄. Submitted an issue.
